app/src/citations.py

```diff
@@ -1,38 +1,63 @@
 import logging
 import random
 import re
-from typing import Match, Sequence
+from itertools import count
+from typing import Callable, Match, Sequence
 
-from src.db.models.document import Chunk, ChunkWithScore, ChunkWithSubsection
+from src.db.models.document import Chunk, ChunkWithSubsection
 from src.util.bem_util import get_bem_url
 
 logger = logging.getLogger(__name__)
 
 _footnote_id = random.randint(0, 1000000)
 _footnote_index = 0
 
-CITATION_PATTERN = r"\(citation-(\d+)\)"
+CITATION_PATTERN = r"\((citation-\d+)\)"
 
 
-def split_into_subsections(
-    chunks: Sequence[Chunk], delimiter: str = "\n\n"
-) -> Sequence[ChunkWithSubsection]:
-    # Given a list of chunks, split them into a flat list of subsections
-    context_mapping = []
+class CitationIdFactory:
```

Comment:
What's the purpose of CitationIdFactory -- to give us flexibility to change ID-generation strategies if we want to move away from `1`, `2`, etc. in the future?

Reply:
Yes, for flexibility. In case we want to start from 0 or 1 or 100, use a different prefix (other than `citation-`), create random IDs, etc.

RE: naming of `CitationIdFactory`. LOL, it was originally `CitationFactory`. All it's doing is assigning a new ID to the citation. Let me ponder this.

Reply:
Since this factory is now the only location where `ChunkWithSubsection` is created, it makes sense to call it `CitationFactory`, esp. since a "subsection" is semantically the same as a "citation".